### PR TITLE
Change simulate methods in PlayerMock to return the event they fired. Change build.gradle to run tests.

### PR DESCRIPTION
**Pull request by ohowe1**
# Description
State the changes of the pull request here.
This pull request changed the simulate methods in `PlayerMock` to return the event that they fired to allow for better testing. This pull request also added the lines:
```
test {
    useJUnitPlatform()
}
```
to the build.gradle file so that the junit tests run when the test task is run. If this was intentional, let me know so that I can roll back the file.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
Before the pull request can be accepted, the version number stored in settings.gradle should be updated.
MockBukkit uses semantic-versioning (https://semver.org/)

 - If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
 - If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)
 - Make sure that unit tests are added which test the relevant changes.
 - Make sure that the changes follow the existing code format.

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

The version number can be found in `settings.gradle`.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 1]
```diff
@@ -283,12 +283,13 @@ public void respawn()
 	 *
 	 * @param moveLocation Location to move player to
 	 */
-	public void simulatePlayerMove(@NotNull Location moveLocation)
+	public PlayerMoveEvent simulatePlayerMove(@NotNull Location moveLocation)
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
	public @NotNull PlayerMoveEvent simulatePlayerMove(@NotNull Location moveLocation)
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 2]
```diff
@@ -972,14 +973,15 @@ public void setSneaking(boolean sneaking)
 		this.sneaking = sneaking;
 	}
 
-	public void simulateSneak(boolean sneak)
+	public PlayerToggleSneakEvent simulateSneak(boolean sneak)
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
	public @NotNull PlayerToggleSneakEvent simulateSneak(boolean sneak)
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 3]
```diff
@@ -994,14 +996,15 @@ public void setSprinting(boolean sprinting)
 		this.sprinting = sprinting;
 	}
 
-	public void simulateSprint(boolean sprint)
+	public PlayerToggleSprintEvent simulateSprint(boolean sprint)
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
	public @NotNull PlayerToggleSprintEvent simulateSprint(boolean sprint)
```

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 4]
```diff
@@ -1538,14 +1541,15 @@ public void setFlying(boolean value)
 		this.flying = value;
 	}
 
-	public void simulateToggleFlight(boolean fly)
+	public PlayerToggleFlightEvent simulateToggleFlight(boolean fly)
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]:
```suggestion
	public @NotNull PlayerToggleFlightEvent simulateToggleFlight(boolean fly)
```

File: build.gradle [hunk 5]
```diff
@@ -133,6 +133,10 @@ publishing {
     }
 }
 
+test {
+    useJUnitPlatform()
```
Comment (TheBusyBiscuit) [inline build.gradle:137]: Afaik the `Junit` *should* be the default, so this shouldn't be necessary.
But it cannot hurt to declare this explicitly. 👍🏻

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks good to me, some missing annotations can be added though

**Comment by ohowe1** [comment]
Those requests should be implemented now. As for Junit, it must have been a local preference that made it so tests didn't run for some reason.

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM.

Thanks for changing that one unit test to a ParameterizedTest and fixing that typo!

**Comment by seeseemelk** [comment]
I'm a little bit considered that the return values of the simulate methods were changed. This introduces a breaking change which could break projects when they attempt to upgrade from `v1.1.0` to `v1.2.0`. I would have preferred to see these being introduced under a slightly different name, and have the old methods be annotated with `@deprecated`. But it has already been merged

**Comment by TheBusyBiscuit** [comment]
@seeseemelk Oh, sorry about that.
We can always change that in retrospect, afaik you haven't pushed it to maven-central yet right?

**Pull request merged**